data/ui/preferences-window.ui

```diff
@@ -2,29 +2,83 @@
 <!-- Generated with glade 3.22.1 -->
 <interface domain="org.gnome.Shell.Extensions.GSConnect">
   <requires lib="gtk+" version="3.20"/>
-  <object class="GtkPopover" id="service-popover">
+  <object class="GtkPopover" id="rename-popover">
     <property name="can_focus">False</property>
+    <signal name="focus-out-event" handler="_onUnfocusServiceName" swapped="no"/>
     <child>
-      <object class="GtkEntry" id="service-entry">
+      <object class="GtkGrid" id="rename">
+        <property name="name">rename</property>
```

Comment:
I don't think `name` is necessary. It's really just some arbitrary text field we sometimes use to identify widgets, but can't really be used to find them like `id`. eg. `id` is what we use in the `Children: []` list.